File: C++/435. Non-overlapping Intervals.cpp
/**
 * Definition for an interval.
 * struct Interval {
 *     int start;
 *     int end;
 *     Interval() : start(0), end(0) {}
 *     Interval(int s, int e) : start(s), end(e) {}
 * };
 */
class Solution {
public:
    int eraseOverlapIntervals(vector<Interval>& intervals) {
        if (intervals.empty())
            return 0;
        sort(intervals.begin(), intervals.end(), [](Interval x, Interval y){
            return x.start < y.start;
        });
        int count = 0;
        int i = 0;
        int j = 1;
        while (i < intervals.size() - 1&& j < intervals.size()){
            if (intervals[i].end > intervals[j].start){
                count++;                                
                if (intervals[j].end < intervals[i].end)
                    i = j;
                j++;
            }
            else{
                i = j;
                j++;
            }
        }    
        return count;
    }
};

File: Python/435. Non-overlapping Intervals.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        if len(intervals) == 0:
            return 0
        intervals = sorted(intervals, key=lambda interval: interval[1])
        end = intervals[0][1]
        count = 1
        for interval in intervals[1:]:
            if interval[0] >= end:
                end = interval[1]
                count += 1
        return len(intervals) - count
        